packages/bitcore-wallet-service/src/lib/fiatrateservice.ts

```diff
@@ -84,28 +84,39 @@ export class FiatRateService {
 
   _retrieve(provider, coin, cb) {
     logger.debug(`Fetching data for ${provider.name} / ${coin} `);
+
+    const handleCoinsRates = (err, res) => {
+      if (err || !res) {
+        return cb(err);
+      }
+
+      logger.debug(`Data for ${provider.name} /  ${coin} fetched successfully`);
+
+      if (!provider.parseFn) {
+        return cb(new Error('No parse function for provider ' + provider.name));
+      }
+      try {
+        const rates = _.filter(provider.parseFn(res), x => _.some(Defaults.FIAT_CURRENCIES, ['code', x.code]));
+        return cb(null, rates);
+      } catch (e) {
+        return cb(e);
+      }
+    };
+
+    const ts = Date.now();
+    if (Constants.BITPAY_USD_STABLECOINS[coin.toUpperCase()]) {
+      return this.getRatesForStablecoin({ code: 'USD', ts }, handleCoinsRates);
+    }
+
+    if (Constants.BITPAY_EUR_STABLECOINS[coin.toUpperCase()]) {
+      return this.getRatesForStablecoin({ code: 'EUR', ts }, handleCoinsRates);
+    }
     this.request.get(
       {
         url: provider.url + coin.toUpperCase(),
         json: true
       },
-      (err, res, body) => {
-        if (err || !body) {
-          return cb(err);
-        }
-
-        logger.debug(`Data for ${provider.name} /  ${coin} fetched successfully`);
-
-        if (!provider.parseFn) {
-          return cb(new Error('No parse function for provider ' + provider.name));
-        }
-        try {
-          const rates = _.filter(provider.parseFn(body), x => _.some(Defaults.FIAT_CURRENCIES, ['code', x.code]));
-          return cb(null, rates);
-        } catch (e) {
-          return cb(e);
-        }
-      }
+      (err, res, body) => handleCoinsRates(err, body)
```

Comment:
Hey @kajoseph, thanks for reviewing. I move it because `handleCoinsRates`  is also used on this lines: 
`return this.getRatesForStablecoin({ code: 'USD', ts }, handleCoinsRates);`
`return this.getRatesForStablecoin({ code: 'EUR', ts }, handleCoinsRates);`